### PR TITLE
Wild card support for log verification

### PR DESCRIPTION
**Pull request by yahman72**
Syntax
- message index wildcard is `*`
- log level wildcard is `ANY`

If the code is okay then I'll update the documentation accordingly

Enhanced also the `test/run.py` to use `robot.output.console.highlighting` to get coloured PASS/FAIL outputs for the statuschecker results

### CODE REVIEW

**Comment by aaltat** [comment]
Fixes #32

**Comment by aaltat** [comment]
Well,  easy fix style things is to use  black and flake8. I can spend some time to implement it in the project. 

I  can review the code too after changes.

**Comment by yahman72** [comment]
yeah, black & flake would be good - I'm so used to them in pre-commit hooks that I don't pay much (any) attention to code style...

File: robotstatuschecker.py [hunk 1]
```diff
@@ -287,31 +287,46 @@ def _get_keyword_rf4(self, test, expected, kw, index):
...
     def _check_message(self, test, kw, expected):
-        try:
-            msg = kw.messages[expected.msg_index]
-        except IndexError:
-            condition = expected.message == "NONE"
-            message = f"Keyword '{kw.name}' (index {expected.kw_index_str}) does not have message {expected.msg_index_str}."
-            self._assert(condition, test, message)
+        if expected.msg_index != "*":
+            try:
+                    msg = kw.messages[expected.msg_index]
```
[the remark this answers is not in the document]
Comment (yahman72) [reply robotstatuschecker.py]: ok = done

File: robotstatuschecker.py [hunk 2]
```diff
@@ -287,31 +287,46 @@ def _get_keyword_rf4(self, test, expected, kw, index):
...
+                self._fail(test, message)
+                return
+
+            for msg in kw.messages:
+                if self._check_msg_message(test, kw, msg, expected, fail=False):
+                    self._check_msg_level(test, kw, msg, expected)
+                    break
+            else:
+                message = f"Keyword '{kw.name}' (index {expected.kw_index_str}) does not contain any logs with message '{expected.message}'"
+                self._fail(test, message)
```
[the remark this answers is not in the document]
Comment (yahman72) [reply robotstatuschecker.py]: ok = done

File: robotstatuschecker.py [hunk 3]
```diff
@@ -287,31 +287,46 @@ def _get_keyword_rf4(self, test, expected, kw, index):
...
+                message = f"Message index wildcard '*' is not supported with expected message 'NONE'"
+                self._fail(test, message)
+                return
+
+            for msg in kw.messages:
+                if self._check_msg_message(test, kw, msg, expected, fail=False):
+                    self._check_msg_level(test, kw, msg, expected)
+                    break
+            else:
+                message = f"Keyword '{kw.name}' (index {expected.kw_index_str}) does not contain any logs with message '{expected.message}'"
```
[the remark this answers is not in the document]
Comment (yahman72) [reply robotstatuschecker.py]: ok = done

**Comment by aaltat** [comment]
It's GitHub safety measure. So that people don't try to get project secrets by submitting malicious code in PR and get the PR automatically build by CI. First time contributors PR build needs approval from maintainers.

**Comment by aaltat** [comment]
There is some conflicts which prevent the merge.

**Pull request merged**